src/v/cluster/metrics_reporter.cc

```diff
@@ -0,0 +1,386 @@
+        if (it == metrics_map.end()) {
+            auto [eit, _] = metrics_map.emplace(
+              report.id, node_metrics{.id = report.id});
+            it = eit;
+        }
+        auto& metrics = it->second;
+
+        metrics.version = report.redpanda_version;
+        metrics.disks.reserve(report.disk_space.size());
+        std::transform(
+          report.disk_space.begin(),
+          report.disk_space.end(),
+          std::back_inserter(metrics.disks),
+          [](const cluster::node_disk_space& nds) {
+              return node_disk_space{.free = nds.free, .total = nds.total};
+          });
+
+        metrics.uptime_ms = report.uptime / 1ms;
+    }
+    auto& topics = _topics.local().topics_map();
+    snapshot.topic_count = 0;
+    snapshot.partition_count = 0;
+    for (const auto& [tp_ns, md] : topics) {
+        // do not include internal topics
+        if (
+          tp_ns.ns == model::redpanda_ns
+          || tp_ns.ns == model::kafka_internal_namespace
+          || !md.is_topic_replicable()) {
+            continue;
+        }
+
+        snapshot.topic_count++;
+        snapshot.partition_count += md.get_configuration().cfg.partition_count;
+    }
+
+    snapshot.nodes.reserve(metrics_map.size());
+    for (auto& [_, m] : metrics_map) {
+        snapshot.nodes.push_back(std::move(m));
+    }
+
+    co_return snapshot;
+}
+
+ss::sstring format_uuid(const std::array<uint8_t, 16> buffer) {
+    std::stringstream ss;
+
+    for (size_t i = 0; i < buffer.size(); ++i) {
+        if (i == 4 || i == 6 || i == 8 || i == 10) {
+            ss << "-";
+        }
+        fmt::print(ss, "{:02x}", buffer.at(i));
+    }
+    return ss.str();
+}
+
+ss::future<> metrics_reporter::try_initialize_cluster_info() {
+    // already initialized, do nothing
+    if (_cluster_uuid != "") {
+        co_return;
+    }
+
+    storage::log_reader_config reader_cfg(
+      model::offset(0), model::offset(2), ss::default_priority_class());
+    reader_cfg.type_filter = model::record_batch_type::raft_configuration;
+    auto reader = co_await _raft0->make_reader(reader_cfg);
+
+    auto batches = co_await model::consume_reader_to_memory(
+      std::move(reader), model::no_timeout);
+    /**
+     * In order to seed the UUID generator we use a hash over first two batches
+     * timestamps and initial raft-0 configuration
+     */
+    if (batches.size() < 2) {
+        co_return;
+    }
+
+    auto& first_cfg = batches.front();
+
+    auto data_bytes = iobuf_to_bytes(first_cfg.data());
+    _creation_timestamp = first_cfg.header().first_timestamp;
+    // use timestamps of first two batches in raft-0 log.
+    for (int i = 0; i < 2; ++i) {
+        auto ts_bytes = iobuf_to_bytes(
+          reflection::to_iobuf(batches[i].header().first_timestamp()));
+        data_bytes.append(ts_bytes.c_str(), ts_bytes.size());
+    }
+
+    std::array<uint32_t, 4> hash{};
+    murmurhash3_x64_128(data_bytes.c_str(), data_bytes.size(), hash.data());
```

Comment:
would a cryptographic hash be preferred?

Reply:
i think that we do not need to make the hash cryptographic safe, hashed data doesn't contain any vulnerable information

Reply:
wouldn't it contain ip addresses and hostnames (e.g. from a configuration batch)?

Reply:
will change this to sha256